src/puppetlabs/puppetdb/utils.clj

```diff
@@ -290,6 +290,21 @@
             (update 1 #(Integer/parseInt (subs % 1 (dec (count %))))))
         s))))
 
+(defn regex-quote
+  [s]
+  (when (and (string? s) (re-find #"\\E" s))
+    (throw (IllegalArgumentException. "cannot regex-quote strings containing '\\E'")))
+  (format "\\Q%s\\E" (str s)))
+
+(defn match-any-of
+  "Given a collection of strings and characters, construct a regex string
+  suitable for passing to re-pattern that consists of a capturing group which
+  matches any member of the collection."
+  [strings]
+  (format "(%s)" (->> strings
+                      (map regex-quote)
+                      (string/join "|"))))
```

Comment:
this is a commit I cherry-picked from @aperiodic I don't have an opinion, but if there was a change to be made, I'd vote for a separate commit that we add on in his PR.